### PR TITLE
Add the template for content authors to easily get started with dotnet try

### PR DESCRIPTION
**Pull request by akshita31**
This PR adds a template which can be installed using a nuget package as follows:

1. dotnet new -i Microsoft.DotNet.Try.ProjectTemplate.Tutorial
(Right now we will have to use fully qulified path to the nupkg so something like
dotnet new -i ./Microsoft.DotNet.Try.ProjectTemplate.Tutorial.1.0.44142.42.nupkg)

This would add the template to the list of templates in dotnet new and the output would be like below:
image

2. Execute `dotnet new trydotnet-tutorial`.

This would create a project file, a Program.cs and a Readme.md which will have reference to the project and the program file. Now you can execute "dotnet try" directly in this repository.

### CODE REVIEW

File: MLS.Agent.Tests/TemplateTests.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+using System.IO;
+using FluentAssertions;
+using System.Threading.Tasks;
+using System.CommandLine;
+using MLS.Agent.CommandLine;
+using MLS.Agent.Tools;
+using Microsoft.DotNet.Try.Protocol.Tests;
+using System.Linq;
+
+namespace MLS.Agent.Tests
```
Comment (jonsequitur) [inline MLS.Agent.Tests/TemplateTests.cs]: These tests should probably go into a separate project specific to the templates.

File: DotNetTry.sln [hunk 2]
```diff
@@ -51,6 +51,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "MLS.HumanizerRunner", "MLS.
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "PackageTool.Tests", "PackageTool.Tests\PackageTool.Tests.csproj", "{11FDD0E8-D07B-41C9-AF7E-E7F735D91ECF}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.DotNet.Try.Template", "Microsoft.DotNet.Try.Template\content\Microsoft.DotNet.Try.Template.csproj", "{9395C81D-C557-4681-8C4C-D5C1F160CD58}"
```
Comment (jonsequitur) [inline DotNetTry.sln]: Re: the project name, will additional templates go into this same project?

Also, "project template" might be more informative than "template".

File: Microsoft.DotNet.Try.ProjectTemplate/content/Program.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            string project = null,
+            string[] args = null)
+        {
+            return region switch
+            {
+                "HelloWorld" => HelloWorld(),
+                "DateTime" => DateTime(),
+                "Guid" => Guid(),
+                "EmptyRegion" => EmptyRegion(),
+                _ => EmptyRegion()
```
Comment (jonsequitur) [inline Microsoft.DotNet.Try.ProjectTemplate/content/Program.cs]: :grinning:

**Comment by akshita31** [comment]
@brettfo Where can I see those logs ?

**Comment by brettfo** [comment]
Click `Details` to the right of the 'try-ci (Windows_NT)' status check then '0 errors / 0 warnings' under the success message, and view the 'Build / Test' step.  Finally in the log there are only 4 instances of ['Successfully created package'](https://dev.azure.com/dnceng/public/_build/results?buildId=198142&view=logs&jobId=63b49790-6233-5870-8977-1ff42860cf98&taskId=42fc5d49-b856-5d4f-b7dc-30a25b6844c8&lineStart=1593&lineEnd=1593&colStart=3&colEnd=31).

**Comment by akshita31** [comment]
https://github.com/tintoy/dotnet-template-templates. I referred this repo for adding the csproj that will include the package id and all the metadata and it seems to work well 🗡

**Comment by akshita31** [comment]
Also the above repo installs templates that looks like

image

**Comment by akshita31** [comment]
@brettfo I am not able to understand how can I download the artifacts produced by this build. I went to the artifacts section but I see only logs there. Do we need to push it to myget to be able to see it ?

**Pull request merged**